internal/test/providers/aws/aws.go

```diff
@@ -0,0 +1,404 @@
+	// so the question marks will match the date of creation
+	// The image obtained by using windowsAMIFilterValue is compatible with  the test container image -
+	// "mcr.microsoft.com/powershell:lts-nanoserver-1809". If the windowsAMIFilterValue changes,
+	// the test container image also needs to be changed.
+	windowsAMIFilterValue := "Windows_Server-2019-English-Full-ContainersLatest-????.??.??"
+	searchFilter := ec2.Filter{Name: &windowsAMIFilterName, Values: []*string{&windowsAMIFilterValue}}
+
+	describedImages, err := ec2Client.DescribeImages(&ec2.DescribeImagesInput{
+		Filters: []*ec2.Filter{&searchFilter},
+		Owners:  []*string{&windowsAMIOwner},
+	})
+	if err != nil {
+		return "", err
+	}
+	if len(describedImages.Images) < 1 {
+		return "", fmt.Errorf("found zero images matching given filter: %v", searchFilter)
+	}
+
+	// Find the last created image
+	latestImage := describedImages.Images[0]
+	latestTime, err := time.Parse(time.RFC3339, *latestImage.CreationDate)
+	if err != nil {
+		return "", err
+	}
+	for _, image := range describedImages.Images[1:] {
+		newTime, err := time.Parse(time.RFC3339, *image.CreationDate)
+		if err != nil {
+			return "", err
+		}
+		if newTime.After(latestTime) {
+			latestImage = image
+			latestTime = newTime
+		}
+	}
+	return *latestImage.ImageId, nil
+}
+
+// getSubnet tries to find a subnet under the VPC and returns subnet or an error.
+// These subnets belongs to the OpenShift cluster.
+func (a *awsProvider) getSubnet(infraID string) (*ec2.Subnet, error) {
+	vpc, err := a.getVPCByInfrastructure(infraID)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get the VPC %v", err)
+	}
+	// search subnet by the vpcid owned by the vpcID
+	subnets, err := a.ec2.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{vpc.VpcId},
+			},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the instance offerings that support Windows instances
+	scope := "Availability Zone"
+	productDescription := "Windows"
+	f := false
+	offerings, err := a.ec2.DescribeReservedInstancesOfferings(&ec2.DescribeReservedInstancesOfferingsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("scope"),
+				Values: []*string{&scope},
+			},
+		},
+		IncludeMarketplace: &f,
+		InstanceType:       &a.instanceType,
+		ProductDescription: &productDescription,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error checking instance offerings of %s: %v", a.instanceType, err)
+	}
+	if offerings.ReservedInstancesOfferings == nil {
+		return nil, fmt.Errorf("no instance offerings returned for %s", a.instanceType)
+	}
+
+	// Finding required subnet within the vpc.
+	foundSubnet := false
+	requiredSubnet := "-private-"
+
+	for _, subnet := range subnets.Subnets {
+		for _, tag := range subnet.Tags {
+			// TODO: find required subnet by checking igw gateway in routing.
```

Comment:
Please create a ticket for this

Reply:
We need a ticket to address this separate from creating a test framework. That ticket can be blocked by the test framework ticket, so that we have fix it in only one spot.

Reply:
https://issues.redhat.com/browse/WINC-491
https://issues.redhat.com/browse/WINC-492

Reply:
linked